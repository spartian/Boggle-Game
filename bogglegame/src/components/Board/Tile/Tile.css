.row button {
    width: 52px;
    height: 52px;
    border-radius: 5px;
    float: left;
    margin-right: 10px;
    margin-bottom: 10px;
    -webkit-transition: background 0.5s; /* Safari */
    transition: background 0.5s;
    outline: none;
    text-transform: uppercase;
}

.tile {
    background: #fff;
    border: solid 1px #757575;
    box-shadow: 2px 2px 5px rgba(255, 204, 128, .25);
    font-size: 24px;
    font-weight: normal;
    text-align: center;
    color: #18252e;
    width: 100%;
    user-select: none;
}

.tile:hover {
    background: #fcf6ea;
    cursor: pointer;
}

.tile-selected {
    background: #FF9800;
    border: solid 1px #FB8C00;
    box-shadow: inset 2px 2px 5px rgba(223, 121, 10, .7);
    width: 100%;
    font-size: 24px;
    font-weight: normal;
    text-align: center;
    color: #ffffff;
}

.tile-selected:hover {
    background: #EF6C00;
    cursor: pointer;
}
