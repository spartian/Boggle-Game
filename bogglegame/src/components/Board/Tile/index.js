import React from 'react';
import './Tile.css';

const Tile = props => {
  // TODO: Destructure Tile props
  const { selected, letter } = props;

function handleOnClickEvent()
{
  alert("Sorry! You need to type the letters. Clicking letters funcitonality is not available for now.");
}


  return (
    <button
      className={selected ? 'tile-selected' : 'tile'} 
      onClick={handleOnClickEvent}
    >
      {letter}
    </button>
  );
};

export default Tile;
