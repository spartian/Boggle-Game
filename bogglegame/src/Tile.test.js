import React from 'react';
import expect from 'expect';
import { shallow, configure,mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import Tile from '../src/components/Board/Tile';
import sinon from 'sinon';

let [handleOnClickEvent] = new Array(1).fill(jest.fn());
const tileComponent = <Tile letter="B" handleClick={handleOnClickEvent} />;


window.alert = jest.fn();
let enzymeWrapper = shallow(tileComponent);




describe('Tile Test 0 ', () => {

    it('a button should be rendered', () => {
        const component = renderer.create(tileComponent);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});




describe('Tile Test 1', () => {
    it('Check if the tile appears or not', () => {
        window.alert.mockClear();
        expect(
            enzymeWrapper.find('button[class="tile"]')).toBeTruthy();
    });
  });


describe('Tile Test 2', () => {
    it('Check if the tile displays same letter', () => {
        window.alert.mockClear();
        
        let button = enzymeWrapper.find('button.tile');
        button.simulate('click');

        // Check if the button text is as expected
        expect(button.text()).toEqual('B');

    });
  });


  