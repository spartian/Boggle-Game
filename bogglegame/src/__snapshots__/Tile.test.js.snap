// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Test 0  a button should be rendered 1`] = `
<button
  className="tile"
  onClick={[Function]}
>
  B
</button>
`;
